src/utils/gocdHelpers.ts

```diff
@@ -91,12 +91,15 @@ export function getProgressColor(pipeline: GoCDPipeline) {
 }
 
 export function firstGitMaterialSHA(
-  deploy: DBGoCDDeployment | undefined
+  deploy?: Pick<DBGoCDDeployment, 'pipeline_build_cause'>
```

Comment:
Are there any downsides to using Pick here? For example, would this not relax TS's type safety checking?

Reply:
Since we only need the `pipeline_build_cause` from the `DBGoCDDeployment`, using `Pick` ensures our type definition remains concise and focused. This approach allows us to work with both supersets types (full `DBGoCDDeployment` types) and more specific types, simplifying testing while maintaining type safety. ([Pick docs](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys))